knotx-core/knotx-template-engine/src/main/java/com/cognifide/knotx/engine/impl/TemplateEngine.java

```diff
@@ -53,15 +56,23 @@ private void initHandlebars() {
         return extractFragments(request.getTemplate())
                 .doOnNext(this::traceSnippet)
                 .flatMap(this::compileHtmlFragment)
-                .concatMapEager(item ->
-                        snippetProcessor.processSnippet(item, request)
+                .filter(htmlFragment -> shouldProcessRequest(htmlFragment, request))
```

Comment:
Yes. this is the point of this issue (#24) to process only the snippets that has the same "data-id" as in AJAX POST form "_id"